File: src/main/java/team4/footwithme/team/service/response/TeamResponse.java
package team4.footwithme.team.service.response;

import team4.footwithme.team.domain.TeamMember;
import team4.footwithme.team.domain.TeamMemberRole;

public record TeamResponse(Long teamMemberId, Long teamId, String name, TeamMemberRole role) {
    public static TeamResponse of(TeamMember teamMember) {
        return new TeamResponse(
            teamMember.getTeamMemberId(),
            teamMember.getTeam().getTeamId(),
            teamMember.getMember().getName(),
            teamMember.getRole()
        );
    }
}

File: src/main/java/team4/footwithme/team/service/response/TeamResponse.kt
package team4.footwithme.team.service.response

import team4.footwithme.team.domain.TeamMember
import team4.footwithme.team.domain.TeamMemberRole

@JvmRecord
data class TeamResponse(val teamMemberId: Long?, val teamId: Long?, @JvmField val name: String?, @JvmField val role: TeamMemberRole?) {
    companion object {
        fun of(teamMember: TeamMember?): TeamResponse {
            return TeamResponse(
                teamMember!!.teamMemberId,
                teamMember.team?.teamId,
                teamMember.member?.name,
                teamMember.role
            )
        }
    }
}
